File: AD-HOC/2382/main.c
#include <math.h>
#include <stdbool.h>
#include <stdio.h>

bool fits(int l, int a, int p, int r) {
    return sqrt(pow(l, 2) + pow(a, 2) + pow(p, 2)) <= r * 2;
}

int main() {
    int l, a, p, r;
    scanf("%d %d %d %d", &l, &a, &p, &r);
    printf("%s\n", fits(l, a, p, r) ? "S" : "N");
    return 0;
}

File: AD-HOC/2382/Main.java
import java.util.Scanner;

public class Main {

    static boolean fits(int l, int a, int p, int r) {
        return Math.sqrt(Math.pow(l, 2) + Math.pow(a, 2) + Math.pow(p, 2)) <= r * 2;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int l = scanner.nextInt();
        int a = scanner.nextInt();
        int p = scanner.nextInt();
        int r = scanner.nextInt();
        System.out.println(fits(l, a, p, r) ? "S" : "N");
        scanner.close();
    }
}
